src/components/Select/components/MultiselectTextField/MultiselectTextField.tsx

```diff
@@ -0,0 +1,204 @@
+  selectedOptions?: SelectOption[];
+  /** Callback fired when the `input` value typed is changed */
+  onInput?: React.EventHandler<any>;
+  /** Value of the `input` element */
+  value?: string | number;
+  /** Callback fired when the Delete button of each Chip is clicked */
+  onOptionDelete: (option?: SelectOption) => void;
+  /** Callback fired when the Delete button of the whole MultiSelect is clicked */
+  onClearAllOptions: () => void;
+  /** If the component is loading */
+  isLoading?: boolean;
+} & Omit<TextFieldProps, 'size'>;
+
+const MultiselectTextField = React.forwardRef<HTMLInputElement, Props & InputProps & TestProps>(
+  (props, ref) => {
+    const {
+      selectedOptions,
+      value,
+      disabled,
+      locked,
+      status,
+      lean,
+      dark,
+      readOnly,
+      label,
+      id,
+      placeholder,
+      required = false,
+      styleType,
+      onOptionDelete,
+      onClearAllOptions,
+      isLoading,
+      rightIcon,
+      ...rest
+    } = props;
+
+    const TextfieldRef = React.useRef<HTMLDivElement>(null);
+
+    const theme = useTheme();
+
+    const hasValue = Boolean(value || (selectedOptions?.length && selectedOptions?.length > 0));
+
+    const inputPlaceholder = useMemo(() => {
+      if (!label && placeholder) {
+        if (required) {
+          return `${placeholder} *`;
+        }
+
+        return placeholder;
+      }
+
+      return label;
+    }, [label, placeholder, required]);
+
+    const iconName = useMemo(() => {
+      if (locked) {
+        return 'lock';
+      }
+
+      if (hasValue) {
+        return 'close';
+      }
+
+      return 'search';
+    }, [hasValue, locked]);
+
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+      if (value === '' && event.key === 'Backspace') {
+        onOptionDelete();
+      }
+    };
+
+    const chips = useMemo(
+      () => (
+        <>
+          {selectedOptions?.map((option, index) => (
+            <span key={generateUniqueID('chip' + index)} css={chipStyle()}>
+              <Chip
+                onClear={!(locked || disabled) ? () => onOptionDelete(option) : undefined}
+                fill="lightGrey"
+                dataTestId={`chip_${index}`}
+              >
+                <div
+                  title={option.label}
+                  css={chipContent({
+                    maxWidth: TextfieldRef.current?.getBoundingClientRect().width,
+                  })}
+                >
+                  {option.label}
+                </div>
+              </Chip>
+            </span>
+          ))}
+        </>
+      ),
+      [disabled, locked, onOptionDelete, selectedOptions]
+    );
+
+    const icon = useMemo(() => {
+      if (rightIcon) {
+        if (typeof rightIcon === 'string') {
+          return <Icon name={rightIcon} size={20} color={theme.utils.getColor('lightGrey', 650)} />;
+        }
+
+        return rightIcon;
```

Comment:
```suggestion
       return typeof rightIcon === 'string' ? <Icon name={rightIcon} size={20} color={theme.utils.getColor('lightGrey', 650)} /> : rightIcon;
```

Reply:
there is another return statement below so the ternary would not be the best option here, as it would increase the complexity of the code